LibAddonMenu-2.0/controls/dropdown.lua

```diff
@@ -360,17 +278,71 @@ function ScrollableDropdownHelper:CalculateContentWidth()
     return maxWidth
 end
 
-function ScrollableDropdownHelper:OnMouseEnter(control)
-    if control.m_data.tooltip then
-        DoShowTooltip(control, control.m_data.tooltip)
+local function AdjustDimensions(control, dropdown, dropdownData)
+    local numItems = #dropdown.m_sortedItems
+    local dropdownObject = dropdown.m_dropdown
+    local scroll = dropdownObject:GetNamedChild("Scroll")
+    local scrollContent = scroll:GetNamedChild("Contents")
+    local anchorOffset = 0
+
+    local isMultiSelectionEnabled = GetDefaultValue(dropdownData.multiSelect)
+
+    local contentWidth = CalculateContentWidth(dropdown) + CONTENT_PADDING
+    local visibleRows, min, max = SetDropdownHeight(control, dropdown, dropdownData)
+
+    if numItems > visibleRows then
```

Comment:
this is really hard to read and understand. I suggest you break this up into multiple sections, e.g. like this:
```lua
local MULTISELECT_NO_SCROLLBAR_PADDING = 6 -- place this next to SCROLLBAR_PADDING above

if isMultiSelectionEnabled then
    anchorOffset = -MULTISELECT_NO_SCROLLBAR_PADDING
end

local hasScrollbar = false
if numItems > visibleRows then
    numItems = visibleRows
    if isMultiSelectionEnabled then
        hasScrollbar = dropdownData.scrollable ~= nil and not scroll.scrollbar:IsHidden()
    else
        hasScrollbar = true
    end
end

if hasScrollbar then
    contentWidth = contentWidth + SCROLLBAR_PADDING
    anchorOffset = -SCROLLBAR_PADDING
end
```